src/metpy/calc/thermo.py

```diff
@@ -4205,6 +4206,223 @@ def k_index(pressure, temperature, dewpoint, vertical_dim=0):
+    >>> # calculate dewpoint
+    >>> Td = dewpoint_from_relative_humidity(T, rh)
+    >>> galvez_davison_index(p, T, Td)
+    <Quantity(-8.06886508, 'dimensionless')>
+    """
+    # Calculate mixing ratio from dewpoint in two steps
+    relative_humidity = relative_humidity_from_dewpoint(
+        temperature, dewpoint
+    )
+    mixing_ratio = mixing_ratio_from_relative_humidity(
+        pressure, temperature, relative_humidity
+    )
+
+    # Calculate potential temperature
+    potential_temp = potential_temperature(pressure, temperature)
+
+    if np.any(np.max(pressure, axis=vertical_dim) < 950 * units.hectopascal):
+        indices_without_950 = np.where(
+            np.max(pressure, axis=vertical_dim) < 950 * units.hectopascal
+        )
+        raise ValueError(
+            f'Data not provided for 950hPa or higher pressure. '
+            f'GDI requires 950hPa temperature and dewpoint data, '
+            f'see referenced paper section 3.d. in docstring for discussion of'
+            f' extrapolating sounding data below terrain surface in high-'
+            f'elevation regions.\nIndices without a 950hPa or higher datapoint'
+            f':\n{indices_without_950}'
+            f'\nMax provided pressures:'
+            f'\n{np.max(pressure, axis=0)[indices_without_950]}'
+        )
+    # Convert temperature before interpolation
+    temperature_k = temperature.to('kelvin')
+    potential_temp_k = potential_temp.to('kelvin')
+
+    # Interpolate to find temperature and mixing ratio at 950, 850, 700, and 500 hPa
+    (
+        (t950, t850, t700, t500),
+        (r950, r850, r700, r500),
+        (th950, th850, th700, th500)
+    ) = interpolate_1d(
+        units.Quantity([950, 850, 700, 500], 'hPa'),
+        pressure, temperature_k, mixing_ratio, potential_temp_k,
+        axis=vertical_dim,
+    )
+
+    th_a = th950
+    r_a = r950
+
+    th_b = 0.5 * (th850 + th700)
+    r_b = 0.5 * (r850 + r700)
+
+    th_c = th500
+    r_c = r500
+
+    alpha = -10 * units.kelvin  # Empirical adjustment
+
+    # Latent heat of vaporization of water - different from MetPy constant
+    #  Using different value, from paper, since GDI unlikely to be used to
+    #  derive other metrics
+    l_0 = 2.69E6 * (units.joule / units.kilogram)
+
+    # Temperature math from here on requires kelvin units
+    eptp_a = th_a * np.exp((l_0 * r_a) / (Cp_d * t850))
+    eptp_b = th_b * np.exp((l_0 * r_b) / (Cp_d * t850)) + alpha
+    eptp_c = th_c * np.exp((l_0 * r_c) / (Cp_d * t850)) + alpha
+
+    if t950.size == 1:
+        is_array = False
+    else:
+        is_array = True
+
+    # Calculate C.B.I.
+    beta = 303 * units.kelvin  # Empirical adjustment
+    l_e = eptp_a - beta  # Low-troposphere EPT
+    m_e = eptp_c - beta  # Mid-troposphere EPT
+
+    # Gamma unit - likely a typo from the paper, should be units of K^(-2) to
+    #  result in dimensionless CBI
+    gamma = 6.5e-2 * (1 / units.kelvin)
```

Comment:
Units on gamma should be 1/K^2.

Also, we'll want to do the multiplication of units as

```units.Quantity(6.5e-2, '1 / kelvin**2')```

This should be done every place there is a multiplication of units.